documentation/2.0/content/userguide/target_env.md

```diff
@@ -24,7 +33,73 @@ This example would apply the `k8s` target type to the discovery result, and plac
+
+This target environment can be applied by providing the command-line argument `-target vz`. It will provide this additional processing:
+
+- The `vz_filter.py` filter will be applied to remove model elements that are not compatible with the Kubernetes environment
+- Variables will be injected into the model for port, host, and URL attributes
+- `lax` validation will be applied for the resulting model
+- Credentials in the model will be replaced with references to Kubernetes secrets, and a script to create those secrets will be produced
+- One additional Kubernetes resource file, `vz-application.yaml`, will be produced, with cluster and data source information derived from the model
+
+#### Generic Kubernetes target
+
+This target environment can be applied by providing the command-line argument `-target k8s`. It will provide this additional processing:
+
+- The `k8s_operator_filter.py` filter will be applied to remove model elements that are not compatible with the Kubernetes environment
+- Variables will be injected into the model for port, host, and URL attributes
+- `lax` validation will be applied for the resulting model
+- Credentials in the model will be replaced with references to Kubernetes secrets, and a script to create those secrets will be produced
+
+### Using secret credentials in the model
+
+If a target environment is configured to use Kubernetes secrets for credential attribute values, each of those values is replaced with a token using the format `@@SECRET:@@ENV:DOMAIN_UID@@<secret-suffix>:<key>`. For example:
+```yaml
+PasswordEncrypted: '@@SECRET:@@ENV:DOMAIN_UID@@-jdbc-generic1:password@@'
+```
+When a domain is created or updated using a model with these tokens, the environment variable `DOMAIN_UID` should be set to the domain's UID, and secrets with corresponding names should have been created. See [Model Tokens]({{< relref "/concepts/model#model-tokens" >}}) for more details about using secret tokens.
+
+The WebLogic admin credentials use a variation of this token format. For example:
+```yaml
+domainInfo:
+    AdminUserName: '@@SECRET:__weblogic-credentials__:username@@'
+    AdminPassword: '@@SECRET:__weblogic-credentials__:password@@'
+```
+The token `__weblogic-credentials__` allows these attributes to reference secrets in a specific location. The `WDT_MODEL_SECRETS_NAME_DIR_PAIRS` environment variable should be set to associate `__weblogic-credentials__` to this location. For example:
+```shell
+WDT_MODEL_SECRETS_NAME_DIR_PAIRS=__weblogic-credentials__=/etc/my-secrets
+```
+See [Model Tokens]({{< relref "/concepts/model#model-tokens" >}}) for more details about using the `WDT_MODEL_SECRETS_NAME_DIR_PAIRS` environment variable.
+
+In WebLogic Kubernetes Operator environments, the environment variable `DOMAIN_UID` is automatically set from the value in the domain resource file. The variable `WDT_MODEL_SECRETS_NAME_DIR_PAIRS` is automatically set to the directory containing WebLogic admin credentials.
+
```

Comment:
Not sure where to put it.  The generated models for target wko is only useful for Model In Image in wko, since the domain is managed in runtime using WDT,  for Domain In Image or PV,  the domain is created before deployment.

Reply:
I updated this to 
```
In WebLogic Kubernetes Operator "Model in Image" environments, 
```
to clarify for now.
I'll add more specific information when we add the non-MII targets later.

Reply:
Because there is no other mention of "Model in Image" in the WDT docs, what I would suggest in your first mention, is to make it a cross-reference to the WKO docs, [Model in Image](https://oracle.github.io/weblogic-kubernetes-operator/userguide/managing-domains/model-in-image/).